### DISCUSSION TITLE
请教个广播模式下如何避免重复消费的问题

### DISCUSSION THREAD

**Discussion by MrZbb** [apache/rocketmq#6619] [Q&A]
我们现在开发有个需求，子节点会订阅生产者广播消息，然后做一些长时间的任务操作。子节点会有扩容缩容的情况，那么如何保证广播模式下每个子节点都是从最新的消费位点消费。
以下是个例子：
假如现在子节点有10个，那么子节点运行期间，是基本不会重复消费的，但是这个时候突然要求其中的一个子节点先不用了关闭，因为广播模式下offset是在消费者本地文件存储的，然后过了一段时间后，又希望这个节点启用，那这个时候这个节点就会把停用的这段时间的没有接收的消息重新消费一次。但是期望的这个节点从最新的开始消费。我想到一种投机取巧的方式是在应用每次启动的时候，删除消费者本地offsets文件，这样是不是可以保证子节点每次都是消费最新的，但是这样的话会不会跟rocketmq广播模式的设计理念有违背，求大佬指导，还是说这种需求下就不应该用rocketmq

**Comment by superhx** [comment]
You could create Consumer with different instanceName, then the new Consumer will consume from newest message.

Or you can create a PR, let broadcast consumer support always consumer from the last

**Reply by MrZbb** [reply]
Thanks. Currently, I am using the PostConstruct method to query the local offsets file of the machine instance during application startup and delete it directly. This does ensure that each time the broadcast mode is used, the problem of old data being re-consumed after a machine is unavailable for a period of time and restarted is resolved. However, I feel that this approach is a bit of a workaround

**Comment by cserwen** [comment]
RocketMQ 客户端会基于之前的消费继续进行消费，要想实现这个功能只能在本地消费记录文件上动手脚：
- 像你这样去删除本地位点记录
- 像 @superhx  描述的那样，使用一个不同的instanceName来该报文件保存路径，最简单的是增加一个时间戳进去，缺点是随着重启次数的增加，消费位点的文件会越来越多

我个人认为，广播消费这种场景还是比较常见的，如果你感兴趣的话可以去提交一个issue来讨论实现这个功能。 @MrZbb
